internal/hammer/testdata/gen.go

```diff
@@ -0,0 +1,184 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// gen is a tool to generate test RSA keys and test certificates for hammer.
+package main
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"flag"
+	"fmt"
+	"math/big"
+	"os"
+	"path"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+const (
+	commonName         = "transparency.dev"
+	organization       = "Transparency.dev"
+	organizationalUnit = "TrustFabric"
+	locality           = "London"
+	state              = "London"
+	country            = "GB"
+)
+
+var (
+	outputPath = flag.String("output_path", "./internal/hammer/testdata/", "Output path for private keys and certificates")
+)
+
+func main() {
+	// Generate a new RSA root CA private key.
+	rootPrivKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		klog.Fatalf("Failed to generate root CA private key: %v", err)
+	}
+	if err := saveRSAPrivateKeyPEM(rootPrivKey, path.Join(*outputPath, "test_root_ca_private_key.pem")); err != nil {
+		klog.Fatalf("Failed to save root CA private key: %v", err)
+	}
+
+	// Generate a new root CA certificate.
+	rootCert, err := rootCACert(rootPrivKey)
+	if err != nil {
+		klog.Fatalf("Failed to generate root CA certificate: %v", err)
+	}
+	if err := saveCertificatePEM(rootCert, path.Join(*outputPath, "test_root_ca_cert.pem")); err != nil {
+		klog.Fatalf("Failed to save root CA certificate: %v", err)
+	}
+
+	// Generate a new RSA intermediate CA private key.
+	intermediatePrivKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		klog.Fatalf("Failed to generate intermediate CA private key: %v", err)
+	}
+	if err := saveRSAPrivateKeyPEM(intermediatePrivKey, path.Join(*outputPath, "test_intermediate_ca_private_key.pem")); err != nil {
+		klog.Fatalf("Failed to save intermediate CA private key: %v", err)
+	}
+
+	// Generate a new Intermediate CA certificate.
+	intermediateCert, err := intermediateCACert(rootCert, rootPrivKey, intermediatePrivKey)
+	if err != nil {
+		klog.Fatalf("Failed to generate intermediate CA certificate: %v", err)
+	}
+	if err := saveCertificatePEM(intermediateCert, path.Join(*outputPath, "test_intermediate_ca_cert.pem")); err != nil {
+		klog.Fatalf("Failed to save intermediate CA certificate: %v", err)
+	}
+
+	// Generate a new RSA leaf certificate signing private key.
+	leafCertPrivateKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		klog.Fatalf("Failed to generate leaf certificate signing private key: %v", err)
+	}
+	if err := saveRSAPrivateKeyPEM(leafCertPrivateKey, path.Join(*outputPath, "test_leaf_cert_signing_private_key.pem")); err != nil {
+		klog.Fatalf("Failed to save leaf certificate signing private key: %v", err)
```

Comment:
What's that going to be used for? Should we re-generate it when new leaf certs are being generated?

Reply:
Let's remove this for now unless there's a specific reason why we need it to be stable across runs.

Reply:
Removed.